src/BackupChain.cs

```diff
@@ -90,13 +91,30 @@ private static IEnumerable<BackupMetadata> MostRecentFullBackup(IEnumerable<Back
     private static IEnumerable<BackupMetadata> NextLogBackup(IEnumerable<BackupMetadata> backups,
       BackupMetadata prevBackup)
     {
-      // also gets all the stripes of the next backup
-      return backups.Where(b => b.BackupType == BackupFileTools.BackupType.Log &&
-                                prevBackup.LastLsn >= b.FirstLsn &&
-                                prevBackup.LastLsn <= b.LastLsn &&
-                                !StripedBackupEqualityComparer.Instance.Equals(prevBackup, b));
+     // also gets all the stripes of the next backup
+     return backups.Where(b => b.BackupType == BackupFileTools.BackupType.Log &&
+                               prevBackup.LastLsn == b.FirstLsn &&
+                               !StripedBackupEqualityComparer.Instance.Equals(prevBackup, b));
     }
 
+    private static IEnumerable<BackupMetadata> FirstLogBackupInChain(IEnumerable<BackupMetadata> backups, 
+     BackupMetadata lastDiff)
+    {
+      var possibleLogs = backups.Where(b => b.BackupType == BackupFileTools.BackupType.Log &&
+                                       lastDiff.LastLsn >= b.FirstLsn &&
+                                       lastDiff.LastLsn <= b.LastLsn &&
+                                       !StripedBackupEqualityComparer.Instance.Equals(lastDiff, b));
+
+      var striped = StripedBackupSet.GetStripedBackupSetChain(possibleLogs);
+
+      if (striped.Count() > 1)
+      {
+       return striped.OrderBy(striped => striped.StripedBackups.First().LastLsn).Last().StripedBackups;
+      }
+
+      return possibleLogs;

```

Comment:
This was the best way I found to fix the issue without larger refactors. Notice how I essentially need to get the stripes from the possible logs in order to verify the true count of backups present. Ideally, the change I outlined [here](https://github.com/factset/AgDatabaseMove/pull/110) would be a better overall fix/change to this class.

Reply:
hmm.  Looks like it would work but seems messy.  Lets try for the approach in your other PR if that works